src/beamlime/config/instrument.py

```diff
@@ -0,0 +1,113 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+from collections.abc import Callable, Iterator, Mapping, Sequence
+from dataclasses import dataclass, field
+from typing import Any
+
+from ess.reduce.streaming import StreamProcessor
+
+from beamlime.config.models import Parameter
+from beamlime.handlers.stream_processor_factory import StreamProcessorFactory
+
+
+class InstrumentRegistry(Mapping[str, 'Instrument']):
+    """
+    Registry for instrument configurations.
+
+    This class is used to register and retrieve instrument configurations
+    based on their names. It allows for easy access to the configuration
+    settings for different instruments.
+
+    Note that in practice instruments are registered only when their module, creating
+    an :py:class:`Instrument`, is imported. Beamlime does currently not import all
+    instrument modules but only the requested one (since importing can be slow). This
+    means that the registry will typically contain only a single instrument.
+    """
+
+    def __init__(self) -> None:
+        self._instruments: dict[str, Instrument] = {}
+
+    def register(self, instrument: Instrument) -> None:
+        """Register an instrument configuration."""
+        if instrument.name in self._instruments:
+            raise ValueError(f"Instrument {instrument.name} is already registered.")
+        self._instruments[instrument.name] = instrument
+
+    def __getitem__(self, name: str) -> Instrument:
+        """Get an instrument by name (implements Mapping)."""
+        return self._instruments[name]
+
+    def __iter__(self) -> Iterator[str]:
+        """Return an iterator over instrument names (implements Mapping)."""
+        return iter(self._instruments)
+
+    def __len__(self) -> int:
+        """Return the number of registered instruments (implements Mapping)."""
+        return len(self._instruments)
+
+    def __contains__(self, name: str) -> bool:
+        """Check if an instrument is registered."""
+        return name in self._instruments
+
+
+instrument_registry = InstrumentRegistry()
+
+
+@dataclass(frozen=True, kw_only=True)
+class Instrument:
```

Comment:
There are some other required variables in the `instrument/<name>.py` files which should be moved into this class, but I will leave that for future work.